packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -518,6 +522,21 @@ export class PendingStateManager implements IDisposable {
             switch (pendingState.type) {
                 case "message":
                     {
+                        if (pendingState.messageType === ContainerMessageType.BlobAttach) {
```

Comment:
Should more of that logic be part of BlobManager? It's a policy we chose to implement, but maybe there are other ways to do so and it would be great to be able to swap another implementation of BlobManager that does it differently.
Also I think it will be easier to test.